src/main/java/org/junitpioneer/jupiter/displaynamegenerator/ReplaceCamelCaseAndUnderscoreAndNumber.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2016-2023 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.displaynamegenerator;
+
+import java.lang.reflect.Method;
+
+import org.junit.jupiter.api.DisplayNameGenerator;
+
+/**
+ * <p>A class extending {@linkplain DisplayNameGenerator.Standard }.</p>
+ *
+ * <p>This extension handles method names with CamelCase, underscore and numbers.</p>
+ *
+ * <p>The aim is to simplify unit test display names. Instead of using this method annotation {@linkplain org.junit.jupiter.api.DisplayName },
+ * we can just use this class annotation {@linkplain org.junit.jupiter.api.DisplayNameGeneration } and use that method annotation if needed.
+ * </p>
+ *
+ * <p>This generator follows 3 rules:</p>
```

Comment:
I would like to highlight this section as the perfect example of something that is both important for a user documentation and the javadoc.

Reply:
updated